src/keygen/participant.rs

```diff
@@ -252,38 +285,51 @@ impl KeygenParticipant {
         let messages = self.broadcast(
             rng,
             MessageType::Keygen(KeygenMessageType::R1CommitHash),
-            com_bytes.clone(),
-            message.id(),
+            com_bytes,
+            sid,
             BroadcastTag::KeyGenR1CommitHash,
         )?;
         Ok(messages)
     }
 
+    /// Handle round one messages from the protocol participants.
+    ///
+    /// In round one, each participant broadcasts its commitment to its public
+    /// key share and a "precommitment" to a Schnorr proof. Once all such
+    /// commitments have been received, this participant will send an opening of
+    /// its own commitment to all other parties.
     #[cfg_attr(feature = "flame_it", flame("keygen"))]
     #[instrument(skip_all, err(Debug))]
     fn handle_round_one_msg<R: RngCore + CryptoRng>(
         &mut self,
+        // XXX the rngs aren't actually needed in the `handle_round_...` methods! Remove?
+        // This does affect the `BroadcastOutput::convert` method.
         rng: &mut R,
         broadcast_message: &BroadcastOutput,
         input: &(),
     ) -> Result<ProcessOutcome<<Self as ProtocolParticipant>::Output>> {
         info!("Handling round one keygen message.");
 
+        // XXX should we have a check that we haven't recieved a round one
+        // message _after_ round one is complete? Likewise for all other rounds.
```

Comment:
FYI it looks like this might be trickier than anticipated, due to how message handling is done. I'm tempted to hold off on any such issue related to this for now (or have a generic issue about revisiting the protocol implementations), since I'm not sure how much effort should go into this if we're anticipating a major refactor in the future.